lib/bundler/plugin/installer.rb

```diff
@@ -12,10 +12,15 @@ class Installer
       autoload :Git,      "bundler/plugin/installer/git"
 
       def install(names, options)
+        check_sources_consistency!(options)
+
         version = options[:version] || [">= 0"]
+
         Bundler.settings.temporary(:lockfile_uses_separate_rubygems_sources => false, :disable_multisource => false) do
           if options[:git]
             install_git(names, version, options)
+          elsif options[:local_git]
+            install_local_git(names, version, options)
```

Comment:
For my own future reference, the appearance of an `elsif` is the start of a smell for me, to replace with a `case/when` but this looks ok for now.